src/Orleans/Serialization/BuiltInTypes.cs

```diff
@@ -1275,6 +1278,34 @@ internal static object CopyType(object obj)
         }
 
         #endregion Type
+    
+        #region CancellationTokenWrapper
+        internal static void SerializeCancellationTokenWrapper(object obj, BinaryTokenStreamWriter stream, Type expected)
+        {
+            var ctw = (CancellationTokenWrapper)obj;
+            var cancelled = ctw.CancellationToken.IsCancellationRequested;
+            if (!cancelled)
+            {
+                CancellationTokenManager.RegisterTokenCallbacks(ctw);
```

Comment:
I just don't want serializer to call back into the runtime logic. Serializer may change, be replaced by another one, ... It should only serialize and have zero side effects.